src/parser/jobs/blm/Leylines.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import {Rule, Requirement} from 'parser/core/modules/Checklist'
+import Module from 'parser/core/Module'
+import {ActionLink} from 'components/ui/DbLink'
+import {i18nMark, Trans} from '@lingui/react'
+
+const LEYLINE_DURATION = 30000
+
+export default class Leylines extends Module {
+	static handle = 'leylines'
+	static i18n_id = i18nMark('blm.leylines.title')
+	static title = 'Ley Lines'
+
+	static dependencies = [
+		'checklist',
+	]
+	_CircleOfPowers = {
+		current: null,
+		history: [],
+	}
+	_LeyLineHistory = []
+
+	constructor(...args) {
+		super(...args)
+		this.addHook('cast', {
+			by: 'player',
+			abilityId: ACTIONS.LEY_LINES.id,
+		}, this._onCastLeyLines)
+		this.addHook('applybuff', {
+			by: 'player',
+			abilityId: STATUSES.CIRCLE_OF_POWER.id,
+		}, this._onApplyCircleOfPower)
+		this.addHook('removebuff', {
+			by: 'player',
+			abilityId: STATUSES.CIRCLE_OF_POWER.id,
+		}, this._onRemoveCircleOfPower)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onCastLeyLines(event) {
+		this.ll = this._LeyLineHistory
+		this.ll.push(event.timestamp)
+
+	}
+
+	_onApplyCircleOfPower(event) {
+		this.cop = this._CircleOfPowers
```

Comment:
What's the reasoning for the assignments here and at the end of the function? I'm pretty sure this is no different from manipulating `this._CircleOfPowers` directly - the assignment should be by reference, so modifying `this.cop` should also modify `this._CircleOfPowers` at the same time.